### PR TITLE
Catch the exception for setting default acl for a file

### PR DESCRIPTION
**Pull request by yuzhu**
this addresses #9116

### CODE REVIEW

**Comment by yuzhu** [comment]
@calvinjia PTAL

**Comment by calvinjia** [comment]
@yuzhu Thanks, could you write a unit test for the new behavior?

**Comment by yuzhu** [comment]
@calvinjia added a test, PTAL. Thanks.

File: core/server/master/src/test/java/alluxio/master/file/FileSystemMasterTest.java [hunk 1]
```diff
@@ -1213,6 +1213,21 @@ public void mountUnmount() throws Exception {
...
+    createFileWithSingleBlock(NESTED_FILE_URI);
+
+    Set<String> newEntries = Sets.newHashSet("default:user::rwx",
+        "default:group::rwx", "default:other::r-x");
+    try {
+      mFileSystemMaster.setAcl(NESTED_FILE_URI, SetAclAction.MODIFY,
+          newEntries.stream().map(AclEntry::fromCliString).collect(Collectors.toList()), context);
+      fail("set default acl for a file should fail");
+    } catch (UnsupportedOperationException e) {
+    }
```
Comment (calvinjia) [inline core/server/master/src/test/java/alluxio/master/file/FileSystemMasterTest.java]: Could you use the expected exception pattern here?

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Comment by calvinjia** [comment]
alluxio-bot, cherry-pick this to branch-2.0 please.

**Pull request merged**